.github/workflows/snackpub.yml

```
@@ -0,0 +1,184 @@
+name: Snackpub
+
+defaults:
+  run:
+    working-directory: snackpub
+
+concurrency:
+  group: snackpub-${{ github.ref }}
+  cancel-in-progress: true
+
+on:
+  workflow_dispatch:
+    inputs:
+      deploy:
+        description: What environment should be deployed
+        type: choice
+        default: no-deploy
+        options:
+          - no-deploy
+          - staging
+          - production
+  pull_request:
+    paths:
+      - .github/actions/setup-google-cloud/**
+      - .github/actions/setup-secrets/**
+      - .github/actions/setup-snackpub/**
+      - .github/workflows/snackpub.yml
+      - snackpub/**
+      - .eslint*
+      - .prettier*
+      - yarn.lock
+  push:
+    branches: [main]
+    paths:
+      - .github/actions/setup-google-cloud/**
+      - .github/actions/setup-secrets/**
+      - .github/actions/setup-snackpub/**
+      - .github/workflows/snackpub.yml
+      - snackpub/**
+      - .eslint*
+      - .prettier*
+      - yarn.lock
+
+jobs:
+  review:
+    runs-on: ubuntu-latest
+    steps:
+      - name: 🏗 Setup repository
+        uses: actions/checkout@v3
+
+      - name: 🏗 Setup secrets
+        uses: ./.github/actions/setup-secrets
+        with:
+          git-crypt-key: ${{ secrets.GIT_CRYPT_KEY }}
+
+      - name: 🏗 Setup snackpub
+        uses: ./.github/actions/setup-snackpub
+
+      - name: 🚨 Lint snackpub
+        run: yarn lint --max-warnings 0
+
+  build:
+    if: ${{ github.event_name == 'pull_request' }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: 🏗 Setup repository
+        uses: actions/checkout@v3
+
+      - name: 🏗 Setup secrets
+        uses: ./.github/actions/setup-secrets
+        with:
+          git-crypt-key: ${{ secrets.GIT_CRYPT_KEY }}
+
+      - name: 🏗 Setup Google Cloud SDK
+        uses: ./.github/actions/setup-google-cloud
+        with:
+          project-id: exponentjs
+          project-zone: us-central1
+          project-cluster: general-central
+          service-key: ${{ secrets.GCLOUD_KEY }}
+
+      - name: 🛠 Build snackpub
+        run: skaffold build --filename snackpub/skaffold.yaml
+        working-directory: ./
+
+  deploy-staging:
+    if: ${{ (github.event.inputs.deploy == 'staging' && github.event_name != 'pull_request') || (github.event_name == 'push' && github.ref == 'refs/heads/main') }}
+    needs: review
+    runs-on: ubuntu-latest
+    environment:
+      name: snackpub-staging
+      url: https://staging.snackpub.expo.io/status
```

We use expo.dev for new domain names. We also use "staging-X" for our staging domain names because of two reasons:
1. Our wildcard TLS certificate between the load balancer and the CDN is for `*.expo.dev` but not `*.snackpub.expo.dev`.
2. It makes the number of subdomain components the same between staging and production.

```suggestion
      url: https://staging-snackpub.expo.dev/status
```

---

i don't have enough permission on expo/snack. let me clarify this after getting enough permission to check this environment usage. maybe we could even drop the environment block.